### ISSUE TITLE
Cannot pass make test

### ISSUE THREAD

**Issue by Ruminateer** [PaddleFlow/ElasticServing#30]
Just take notes here
Current main branch 0ae1ea2b00647efcecf034b6a6f286fc2c35fa26 cannot pass `go test` in `make test`
Seems to be bugs in test files but not sure yet.

**Comment by Ruminateer** [comment]
```
$ make test
/home/zzy/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
/home/zzy/go/bin/controller-gen "crd:maxDescLen=0" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
go test ./... -coverprofile cover.out
?       ElasticServing  [no test files]
?       ElasticServing/pkg/apis [no test files]
?       ElasticServing/pkg/apis/elasticserving/v1       [no test files]
?       ElasticServing/pkg/constants    [no test files]
?       ElasticServing/pkg/controllers  [no test files]
ok      ElasticServing/pkg/controllers/elasticserving   3.989s  coverage: 0.0% of statements
?       ElasticServing/pkg/controllers/elasticserving/reconcilers/istio [no test files]
?       ElasticServing/pkg/controllers/elasticserving/reconcilers/knative       [no test files]
--- FAIL: TestCreateVirtualService (0.00s)
    --- FAIL: TestCreateVirtualService/test_case_1 (0.00s)
        virtualservice_test.go:123: Test "test case 1" unexpected service (-want +got):   &v1alpha3.VirtualService{
                TypeMeta:   {Kind: "VirtualService", APIVersion: "networking.istio.io/v1alpha3"},
                ObjectMeta: {Name: "testresource", Namespace: "test"},
                Spec: v1alpha3.VirtualService{
                        Hosts:    {"*"},
                        Gateways: {"paddleflow/paddleflow-gateway"},
                        Http: []*v1alpha3.HTTPRoute{
                                &{
                                        Name:  "",
            -                           Match: []*v1alpha3.HTTPMatchRequest{s`uri:<prefix:"/paddlepaddle/test/testresource/" > `},
            +                           Match: nil,
                                        Route: []*v1alpha3.HTTPRouteDestination{
                                                &{
                                                        Destination: &v1alpha3.Destination{
            -                                                   Host:                 "testresource.test.svc.cluster.local",
            +                                                   Host:                 "testresource-service",
                                                                Subset:               "",
            -                                                   Port:                 s"number:80 ",
            +                                                   Port:                 nil,
                                                                XXX_NoUnkeyedLiteral: {},
                                                                XXX_unrecognized:     nil,
                                                                XXX_sizecache:        0,
                                                        },
                                                        Weight:  0,
                                                        Headers: nil,
                                                        ... // 3 identical fields
                                                },
                                        },
                                        Redirect: nil,
                                        Delegate: nil,
            -                           Rewrite:  s`uri:"/paddlepaddle/test/testresource/" `,
            +                           Rewrite:  nil,
                                        Timeout:  nil,
                                        Retries:  nil,
                                        ... // 9 identical fields
                                },
                        },
                        Tls: nil,
                        Tcp: nil,
                        ... // 4 identical fields
                },
                Status: {},
              }
FAIL
coverage: 80.0% of statements
FAIL    ElasticServing/pkg/controllers/elasticserving/resources/istio   0.024s
--- FAIL: TestPaddleServiceToKnativeService (0.00s)
    service_test.go:104: Test "Test1" unexpected canary service (-want +got):   &v1.Service{
                TypeMeta:   {},
                ObjectMeta: {Name: "test-service", Namespace: "default"},
                Spec: v1.ServiceSpec{
                        ConfigurationSpec: v1.ConfigurationSpec{
                                Template: v1.RevisionTemplateSpec{
                                        ObjectMeta: v1.ObjectMeta{
                                                ... // 9 identical fields
                                                DeletionGracePeriodSeconds: nil,
                                                Labels:                     {"PaddleService": "paddlesvc"},
        -                                       Annotations:                nil,
        +                                       Annotations: map[string]string{
        +                                               "autoscaling.knative.dev/class":                       "kpa.autoscaling.knative.dev",
        +                                               "autoscaling.knative.dev/maxScale":                    "10",
        +                                               "autoscaling.knative.dev/metric":                      "concurrency",
        +                                               "autoscaling.knative.dev/minScale":                    "1",
        +                                               "autoscaling.knative.dev/panicThresholdPercentage":    "200",
        +                                               "autoscaling.knative.dev/panicWindowPercentage":       "10",
        +                                               "autoscaling.knative.dev/target":                      "100",
        +                                               "autoscaling.knative.dev/targetUtilizationPercentage": "70",
        +                                               "autoscaling.knative.dev/window":                      "60s",
        +                                       },
                                                OwnerReferences: nil,
                                                Finalizers:      nil,
                                                ... // 2 identical fields
                                        },
                                        Spec: v1.RevisionSpec{
                                                PodSpec: v1.PodSpec{
                                                        Volumes:        nil,
                                                        InitContainers: nil,
                                                        Containers: []v1.Container{
                                                                {
                                                                        Name:       "latest",
                                                                        Image:      "hub.baidubce.com/paddlepaddle/serving:latest",
        -                                                               Command:    nil,
        +                                                               Command:    []string{"/bin/bash", "-c"},
        -                                                               Args:       nil,
        +                                                               Args:       []string{""},
                                                                        WorkingDir: "",
                                                                        Ports:      {{Name: "http1", ContainerPort: 9292, Protocol: "TCP"}},
                                                                        ... // 4 identical fields
                                                                        VolumeDevices:            nil,
                                                                        LivenessProbe:            nil,
        -                                                               ReadinessProbe:           nil,
        +                                                               ReadinessProbe:           s"&Probe{Handler:Handler{Exec:nil,HTTPGet:nil,TCPSocket:&TCPSocketAction{Port:{0 0 },Host:,},},InitialDelaySeconds:0,TimeoutSeconds:1,PeriodSeconds:100,SuccessThreshold:1,FailureThreshold:100,}",
                                                                        StartupProbe:             nil,
                                                                        Lifecycle:                nil,
                                                                        TerminationMessagePath:   "",
                                                                        TerminationMessagePolicy: "",
        -                                                               ImagePullPolicy:          "Always",
        +                                                               ImagePullPolicy:          "IfNotPresent",
                                                                        SecurityContext:          nil,
                                                                        Stdin:                    false,
                                                                        ... // 2 identical fields
                                                                },
                                                        },
                                                        EphemeralContainers: nil,
                                                        RestartPolicy:       "",
                                                        ... // 29 identical fields
                                                },
        -                                       ContainerConcurrency: nil,
        +                                       ContainerConcurrency: &0,
        -                                       TimeoutSeconds:       nil,
        +                                       TimeoutSeconds:       &300,
                                        },
                                },
                        },
                        RouteSpec: {},
                },
                Status: {},
          }
FAIL
coverage: 72.6% of statements
FAIL    ElasticServing/pkg/controllers/elasticserving/resources/knative 0.036s
FAIL
make: *** [Makefile:18: test] Error 1
```

**Comment by jinminhao** [comment]
Solved in !(Issue#37)[https://github.com/PaddleFlow/ElasticServing/pull/37]

**Comment by Ruminateer** [comment]
> Solved in [PR#37](https://github.com/PaddleFlow/ElasticServing/pull/37)

You can write "Fixes #30" in PR so that when PR is merged the issue is automatically closed 🤗
